models/cm1/model_mod.f90

```diff
@@ -1127,6 +1129,37 @@ function observation_on_grid(obs_location, ndims)
       return ! exit early
 
    endif
+elseif ( periodic_x) then
+    ! require that the point is contained within the staggered grid for the 
+    ! y - direction since you cannot wrap-arround
+    if ( (obs_location(1) < xf(1)) .or. (obs_location(1) > xf(nip1)) .or. &
+        (obs_location(2) < yh(1)) .or. (obs_location(2) > yh(njp1)) ) then
+
+      if (debug > 0) then
+         print *, 'periodic boundary conditions - in x-direction'
+         print *, 'OBSERVATION_x,y at ', obs_location(1:2), ' is off x,y grid'
+         print *, 'x_min, x_max ', xf(1), xf(nip1)
+         print *, 'y_min, y_max ', yh(1), yh(nj)
+      endif
+
+      return ! exit early
+    endif
+elseif ( periodic_y) then
+    ! require that the point is contained within the staggered grid for the 
+    ! x - direction since you cannot wrap-arround
+    if ( (obs_location(1) < xh(1)) .or. (obs_location(1) > xh(nip1)) .or. &
```

Comment:
Same as above comment: xh is lenght ni:

```suggestion
    if ( (obs_location(1) < xh(1)) .or. (obs_location(1) > xh(ni)) .or. &
```

or swap xh -> xf, yf -> yh

Reply:
Same here, the updates look good.